pkg/operator/ceph/disruption/clusterdisruption/pools.go

```diff
@@ -121,30 +123,48 @@ func (r *ReconcileClusterDisruption) reconcileCephObjectStore(cephObjectStoreLis
 			break
 		}
 		blockOwnerDeletion := false
-		pdb := &policyv1beta1.PodDisruptionBudget{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      pdbName,
-				Namespace: namespace,
-				OwnerReferences: []metav1.OwnerReference{
-					{
-						APIVersion:         objectStore.APIVersion,
-						Kind:               objectStore.Kind,
-						Name:               objectStore.ObjectMeta.Name,
-						UID:                objectStore.UID,
-						BlockOwnerDeletion: &blockOwnerDeletion,
-					},
+		objectMeta := metav1.ObjectMeta{
+			Name:      pdbName,
+			Namespace: namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         objectStore.APIVersion,
+					Kind:               objectStore.Kind,
+					Name:               objectStore.ObjectMeta.Name,
+					UID:                objectStore.UID,
+					BlockOwnerDeletion: &blockOwnerDeletion,
 				},
 			},
+		}
+		usePDBV1, err := mon.UsePDBV1Version(r.context.ClusterdContext.Clientset)
+		if err != nil {
+			return err
+		}
+		if usePDBV1 {
+			pdb := &policyv1.PodDisruptionBudget{
+				ObjectMeta: objectMeta,
+				Spec: policyv1.PodDisruptionBudgetSpec{
+					Selector:     labelSelector,
+					MinAvailable: minAvailable,
+				},
+			}
+			request := types.NamespacedName{Name: pdbName, Namespace: namespace}
+			err = r.reconcileStaticPDB(request, pdb)
+			if err != nil {
+				return errors.Wrapf(err, "failed to reconcile cephobjectstore pdb %v", request)
+			}
+		}
```

Comment:
Looks like you need to `return` before the end of this code block, or else you will create both v1 and v1beta1 pdbs.